Active records and migrations

maksymstoliarov opened the pull request.

Commit:
Pretty Url

Beautifull links in menu

Commit:
Added migrations

Added 5 migrations. Tables:students, homework, department, thema and junction table student_homework. Tables students and department have one-to-many relashionship(one department has many students), tables homework and thema have one-to-one relationship(one homework has one thema), tables students and homework have many-to-many relationship(many students have many homeworks)

Commit:
Generated CRUD

Generated CRUD for all tables

Commit:
Test actions

Fixed test actions with using find(), delete(), findAll(), save()

Commit:
Test view table

Changed view for test index to table

dchabanenko commented on controllers/ThemaController.php:
```diff
@@ -0,0 +1,124 @@
+/**
+ * ThemaController implements the CRUD actions for Thema model.
+ */
+class ThemaController extends Controller
```
TopicController? or SubjectController

dchabanenko commented on models/Students.php:
```diff
@@ -0,0 +1,76 @@
+ * @property StudentsHomework[] $studentsHomeworks
+ * @property Homework[] $homeworks
+ */
+class Students extends \yii\db\ActiveRecord
```
Student, not Students. The table can be named "students" or "student" but the model should be "Student"

dchabanenko commented on migrations/m161204_150439_create_department_table.php:
```diff
@@ -0,0 +1,28 @@
+    {
+        $this->createTable('department', [
+            'id' => $this->primaryKey(),
+            'department_name' => $this->string(50)->notNull(),
```
You don't need to add "department_" prefix, it's enough to call this field "name"

dchabanenko commented on models/StudentsHomework.php:
```diff
@@ -0,0 +1,65 @@
+ * @property Homework $homework
+ * @property Students $students
+ */
+class StudentsHomework extends \yii\db\ActiveRecord
```
If this model is only for creating many-to-many relations - you can not create it.

dchabanenko commented on models/Test.php:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    public static function createData(){
+        $new = new Test();
```
Models should not contain methods for testing. Please move the methods createData and editData to controller and make it work as actions.

Commit:
Fixed test form

Changed views of update and create, added basic form, changed students-howework link, added test links, changed update and create test actions, changed buttons for test CRUD
